### ISSUE TITLE
Indentation of numbered lists always uses the same style

### ISSUE THREAD

**Issue by jswiderski** [ckeditor/ckeditor5#12179]
## 📝 Provide detailed reproduction steps (if any)

1. Go to https://ckeditor.com/docs/ckeditor5/latest/features/lists/lists.html#list-styles 
2. Clear editor contents and insert list with few indentation steps

### ✔️ Expected result

For ech sub-list/indentation step, new list style should be used.

### ❌ Actual result

Same style is used for each sub-list/indentation step. 
<img width="496" alt="Screenshot 2022-07-28 at 14 56 06" src="https://user-images.githubusercontent.com/5314687/181510648-ec50ee7d-f433-4d2f-86df-0aababd902a9.png">

Bulleted list changes its styles nicely while Numbered list always uses the first one.

### ❓ Possible solution

_If you have ideas, you can list them here. Otherwise, you can delete this section._

## 📃 Other details

* Browser: Any
* OS: Any
* First affected CKEditor version: N/A
* Installed CKEditor plugins: list

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
@oleq Am I right that this could resolved by a simple set of content styles (`.ck-content ul ul`)?

I'm worried how it's gonna play with the list properties feature – namely: if the actual style is affected by a stylesheet – how will the UX when using list properties work here? 

The list properties feature doesn't mark the active option in the dropdown panel even if it matches the actual content style.

With no specific style applied (stylesheet defines the styling):

![](https://user-images.githubusercontent.com/156149/187399774-bb695a57-4224-451c-a6a4-bfcc12e3d3c7.png)

That's because you need to specifically choose the style. If the user actually chose the style it is active:

![](https://user-images.githubusercontent.com/156149/187400073-aa25e225-0e90-4687-93b8-81a8dae830a8.png)

So, by defining styles for nested lists in the stylesheet we won't degrade anything here. It's going to be still the same. Do you agree?

If so, is this ticket as simple to resolve as I think?

**Comment by oleq** [comment]
> Bulleted list changes its styles nicely while Numbered list always uses the first one.

It does change the styles in CSS but this is because UA specifies them

![](https://user-images.githubusercontent.com/1099479/187416178-c2eb79e0-c96c-4faf-b670-698754e07397.png)

they are not reflected in the styles drop-down of CKEditor 5, though:

![](https://user-images.githubusercontent.com/1099479/187415928-49e735a2-b777-4b99-a2fc-01aef50a9581.png)

The UA (in this case: Chrome) does not bring similar styles for `<ol>`. If we added them for `.ck-content` (`ol ol`, `ol ol ol`, etc.), the styles drop-down would not recognize them either because it reads HTML attributes only.

If we're OK with this drawack, then this is a very simple fix. 

---

But... to me, this whole thing feels like an issue, but not the one submitted by @jswiderski. 

We don't have smart style assignment while changing indentation so let's not pretend we do by bringing confusing content styles. I'd rather reset `ul ul`, `ul ul ul`, `ol ol`, `ol ol ol`, etc. content styles to the sane default value to make sure they are predictable and consistent with the UI. 

End-users may not understand why what they see does not correspond to the state of the UI. They are using a WYSIWYG editor, they don't have to understand UAs, CSS and stuff like that. To them, this will be an **obvious bug**. Besides, other word processors like MS Word are very clear anyway

![](https://user-images.githubusercontent.com/1099479/187417565-caeaa2a8-313c-4a23-a0c1-505a19e6e339.png)

To sum up:

*   Right now, I'd reset all nested list styles to sane default (both numbered and bulleted) to make the feature at least predictable and consistent.
*   (In the future) I would consider automatic style assignment when indenting in the `ListStyle` plugin (by assigning the attribute value; could be an opt-out flag).
*   (In the distant future) I would consider making the `ListStyle` feature smart enough to read computed styles.

**Comment by Reinmar** [comment]
> If we're OK with this drawack, then this is a very simple fix.

I'd say we are OK. I didn't notice this initially, but bulleted lists already have each level styled differently and this ticket is just about the numbered lists. In both cases the UI doesn't reflect the actual computed style and somehow that's enough. 

https://user-images.githubusercontent.com/156149/187524781-e595b43d-bfb3-4757-87d8-66bfadcbc4dd.mp4

So, if there are no technical problems with that, let's just make numbered lists behave like bulleted – this is – have each level styled differently.

In fact, another argument to just do this is that people may not have the list properties plugin loaded and they won't even care what the style type is as long as the content look good. And for that – each level should have different styles.

As for the bigger picture that you touched in your comment – this topic is at least 100x bigger than the originally reported issue and far beyond its scope. We shouldn't burry down a simple fix because of this.

**Comment by oleq** [comment]
The styles needed for ordered lists could be as follows

```css
.ck-content ol {
  list-style-type: decimal;
}

.ck-content ol ol {
  list-style-type: lower-latin;
}

.ck-content ol ol ol {
  list-style-type: lower-roman;
}

.ck-content ol ol ol ol {
  list-style-type: upper-latin;
}

.ck-content ol ol ol ol ol {
  list-style-type: upper-roman;
}
```

And for unordered lists (just make sure we don't rely on UA styles):

```css
.ck-content ul {
  list-style-type: circle;
}

.ck-content ul ul {
  list-style-type: disc;
}

.ck-content ul ul ul {
  list-style-type: square;
}

.ck-content ul ul ul ul {
  list-style-type: square;
}
```

and that should result in the following look:

![](https://user-images.githubusercontent.com/1099479/187649565-d7893793-5238-406a-9bbf-5b50441d302e.png)

**Comment by saahiln2411** [comment]
Hi @oleq @Reinmar 

Please let me know by when can I expect this to be fixed?

**Comment by scofalik** [comment]
It would be good to extract the problems discussed here to separate issue. This will be closed with the simple solution provided by @oleq.

**Comment by spacelist-ca** [comment]
@oleq  What ever happened with this exactly? A bit hard to follow the thread, but the issue of the numbered list itself not having the same styling as the list items. Is that something that is being resolved or has a known fix?
